src/shell.js

```diff
@@ -290,7 +290,7 @@ if (ENVIRONMENT_IS_SHELL) {
 if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
   if (ENVIRONMENT_IS_WORKER) { // Check worker, not web, since window could be polyfilled
     scriptDirectory = self.location.href;
-  } else if (document.currentScript) { // web
+  } else if (typeof document !== 'undefined' && document.currentScript) { // web
```

Comment:
does it make sense for code size to write `if (document && document.currentScript)`?   Or is the semantic difference salient here?  (or does minification do that for us?)

Reply:
There is a semantic difference, yes - `if (document` will throw if `document` is not defined at all in the scope. `typeof` is the way to check for that without throwing.